RELEASES.md

```diff
@@ -0,0 +1,69 @@
+# Release Process
+
+## Active development
+
+Active development is happening on the `master` branch, and a new version is released from it
+at the end of each quarter.
+
+## Stable releases
+
+Stable releases of Envoy include:
+
+* Extended maintenance window (any version released in the last 12 months).
+* Security fixes backported from the `master` branch (including those deemed not worthy CVE).
+* Stability fixes backported from the `master` branch (anything that can result in a crash,
+  including crashes triggered by a trusted control plane).
+* Bugfixes, deemed worthwhile by the maintainers of stable releases.
+
+### Hand-off
+
+Hand-off to the maintainers of stable releases happens after Envoy maintainers release a new
+version from the `master` branch by creating a `vX.Y.0` tag and a corresponding `vX.Y` branch,
+with merge permissions given to the release manager of stable releases, and CI configured to
+execute tests on it.
```

Comment:
Do we know how to do these two things? I think I know how to do the first one (I think it's possible to give push permissions on a branch basis but I'm not 100% positive). How do we run CI on the branches? cc @lizan

Reply:
Yes I can configure AZP to run on specific branch pattern, i.e. `release/*`.